builtin/logical/nomad/path_roles.go

```diff
@@ -0,0 +1,159 @@
+package nomad
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathListRoles(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "role/?$",
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ListOperation: b.pathRoleList,
+		},
+	}
+}
+
+func pathRoles(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "role/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the role",
+			},
+
+			"policies": &framework.FieldSchema{
+				Type:        framework.TypeCommaStringSlice,
+				Description: "Comma separated list of policies as previously created in Nomad. Required",
+			},
+
+			"global": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "Boolean value describing if the token should be global or not. Defaults to false",
+			},
+
+			"type": &framework.FieldSchema{
+				Type:    framework.TypeString,
+				Default: "client",
+				Description: `Which type of token to create: 'client'
+or 'management'. If a 'management' token,
+the "policies" parameter is not required.
```

Comment:
This now says that policies *isn't* always required. You should update the text for policies to indicate that it's required if the type is client.